modelling/src/main/java/org/axonframework/modelling/command/AggregateAnnotationCommandHandler.java

```diff
@@ -418,6 +419,11 @@ public Object handle(CommandMessage<?> command) throws Exception {
             Aggregate<T> aggregate = repository.newInstance(factoryMethod, agg -> {
                 try {
                     response.set(agg.handle(command));
+                } catch (NullLockIdentifierException e) {
```

Comment:
I would go for the option of returning a "NoLock" in the LockingRepository where it creates the LockAwareAggregate. It can be considered as being "not held". In turn (and that change is already included), the "prepareForCommit" for an aggregate isn't invoked if the aggregate doesn't have any changes to commit anyway.

Reply:
Yep, good point. The `AbstractRepository` does a similar check later to validate if the aggregate has a workable identifier, and doesn't prepare for commit without it.
Thanks for the insights :+1: